docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,315 @@
+
+In the detailed design, we will focus on bucket-level immutability policies due to their broader support and simpler management.
+
+### Detailed Design
+
+#### Bucket Immutability Mechanism
+
+The bucket immutability feature configures an immutability policy for a cloud storage bucket, dictating how long objects in the bucket must remain immutable. It also allows for locking the bucket's immutability policy, permanently preventing the policy from being reduced or removed.
+
+- **Supported by Major Providers:**
+  - **Google Cloud Storage (GCS):** [Bucket Lock](https://cloud.google.com/storage/docs/bucket-lock)
+  - **Amazon S3 (S3):** [Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html)
+  - **Azure Blob Storage (ABS):** [Immutable Blob Storage](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-policy-configure-container-scope?tabs=azure-portal)
+
+- **Not Supported:**
+  - **OpenStack Swift**
+
+**Implementation Details:**
+
+- Operators are responsible for configuring new or existing buckets with these immutability settings before `etcd-backup-restore` begins uploading snapshots.
+- Once configured, snapshots uploaded by `etcd-backup-restore` will also be immutable and cannot be altered or deleted until the immutability period expires.
+- No additional configuration needs to be passed to etcd-druid.
+
+#### ETCD Backup Configuration
+
+Operators must ensure that the ETCD backup configuration aligns with the immutability requirements, including setting appropriate immutability periods.
+
+#### Handling of Hibernated Clusters
+
+When an ETCD cluster is hibernated for a duration exceeding the immutability period, backups may become mutable again (this behavior depends on the cloud provider; refer to [Comparison of Storage Provider Properties](#comparison-of-storage-provider-properties-for-bucket-level-and-object-level-immutability)), compromising the intended immutability guarantees.
+
+Such handling of hibernated clusters is the type of scenario which the etcd operator-tasks frameworks lends itself to quite well, and thus for all proposed solutions, the operator tasks framework as defined [here](./05-etcd-operator-tasks.md) will be made use of for the designs of the solutions.
+
+To maintain snapshot immutability during extended hibernation, we propose two approaches:
+
+##### Approach 1: Using the Compaction Job
+
+**Proposed Solution:**
+
+Utilize the compaction job to periodically take fresh snapshots during hibernation. Introduce a new flag `--hibernation-snapshot-interval` to the compaction controller. This flag sets the interval after which a compaction job should be triggered for a hibernated ETCD cluster, based on the time elapsed since `fullLease.Spec.RenewTime.Time` and if `etcd.spec.replicas` is `0` (indicating hibernation). The compaction job uses the [compact command](https://github.com/gardener/etcd-backup-restore/blob/master/cmd/compact.go) to create a new snapshot.
```

Comment:
Shouldn't this be a derived information, based on the last full snapshot time and the etcd.spec.replicas being `0`. 
We already have this information so why not use that.

Reply:
We have removed the compaction approach as mentioned [here](https://github.com/gardener/etcd-druid/pull/884#discussion_r1871190971). 

Anyways, controller periodically creates the `ExtendEtcdSnapshotImmutabilityTask` if `etcd.spec.backup.store.immutability.retentionType` is set to `"Bucket"` and based on `etcd.spec.backup.fullSnapshotSchedule`.